static/docs/install/docker.md

```diff
@@ -0,0 +1,114 @@
+# Using Docker
+
+Currently we don't provide any official Docker image, however you can easily
+create it yourself.
+
+## Writing the Dockerfile
+
+We support several ways to install DVC, the two recommended ones are through
+`pip` and `conda`.
+
+You can base your image on the official Python one:
+
+```dockerfile
+FROM python
+
+# Remember to include the cloud providers that you plan to use,
+# for example: "dvc[s3]"
+RUN pip install dvc
+```
+
+Or use a Conda based image:
+
+```dockerfile
+FROM continuumio/miniconda3
+
+RUN conda install dvc -y --quiet
+```
+
+For a recommended reading about writing Dockerfiles for Python applications, you
+can take a look at https://pythonspeed.com/docker/
+
+## Using Docker for your development environment
+
+Since `dvc` is a command line application, we have several recommendations to
+enhance the experience:
+
+- Make sure your _locale_ is set to UTF-8
+- Install the Zsh shell and its respective autocompletion script.
+- Install `less` so you can pipe the output of `dvc pipeline show`.
+- Enable movements with `Ctrl + Arrow` by editing the _inputrc_ file.
+- Use `pip install dvc` at the end of the file for better caching.
+- Set your `WORKDIR` to the path where you are going to mount your project.
+
+If you are using Linux, remember to map your user to your container. This way,
+you make sure your files are still writable. For more information:
+https://docs.docker.com/engine/security/userns-remap/
+
+### Example Dockerfile
+
+The following is a **suggestion** for a Dockerfile to use during development:
+
+```dockerfile
+FROM python:3.7
+
+# Use UTF-8 as the default locale
+ENV LANG=C.UTF-8
+
+# Better movement with [Ctrl + →] and [Ctrl + ←]
+ENV INPUTRC=/etc/inputrc
+
+RUN set -ex \
+  && echo '"\e[1;5C": forward-word' >> /etc/inputrc \
+  && echo '"\e[1;5D": backward-word' >> /etc/inputrc
+
+# Set working directory.
+# Mount your project under the same path.
+WORKDIR /usr/src
+
+# Install useful programs
+RUN apt-get update -y && apt-get install -y \
+      zsh \
+      less \
+      git
+
+#  Set Zsh as the default shell
+SHELL ["/usr/bin/zsh", "-c"]
```

Comment:
Why change the default? (Bash)

Reply:
I assume to accommodate the custom `zsh` syntax config... Speaking of which, it would be nice to provide equivalent `bash` config if possible

Reply:
Which goes to my point. Probably an unnecessary complication for a Dockerfile sample.

Reply:
@casperdcl , `bash` is not as customizable as `zsh`, completion is really basic :grimacing:

Reply:
I would agree given the argument about providing a better experience, but I think most people are used to Bash, so not sure. It's not like we recommend Zsh over Bash in docs. This one may be unnecessary.

Reply:
Moved to https://github.com/iterative/dvc/pull/2844#pullrequestreview-322660534